### ISSUE TITLE
Getting NU1202  Package Microsoft.NETCore.App 2.0.9 is not compatible with netstandard2.0

### ISSUE THREAD

**Issue by karelz** [dotnet/linker#446]
_From @Petermarcu on September 11, 2018 20:18_

@MihaMarkic commented on [Tue Sep 11 2018](https://github.com/dotnet/core/issues/1929)

# Issue Title

When publishing project, I'm getting some NU1202  errors:
```
...\AutoMasshTik.Engine.csproj : error NU1202: Package Microsoft.NETCore.App 2.0.9 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0). Package Microsoft.NETCore.App 2.0.9 supports: [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
...\AutoMasshTik.Engine.csproj : error NU1202:   - netcoreapp (.NETCoreApp,Version=v0.0) [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
...\AutoMasshTik.Engine.csproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
...\AutoMasshTik.Engine.csproj : error NU1202: Package Microsoft.NETCore.App 2.0.9 is not compatible with netstandard2.0 (.NETStandard,Version=v2.0) / win10-x86. Package Microsoft.NETCore.App 2.0.9 supports: [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
...\AutoMasshTik.Engine.csproj : error NU1202:   - netcoreapp (.NETCoreApp,Version=v0.0) [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
...\AutoMasshTik.Engine.csproj : error NU1202:   - netcoreapp2.0 (.NETCoreApp,Version=v2.0) [...\AutoMasshTik\src\AutoMasshTik.UI\AutoMasshTik.UI.csproj]
```

# General

The project in question is [AutoMasshTik](https://github.com/MihaMarkic/AutoMasshTik). 
I invoke something like 
`
 dotnet publish -o somepath\files  -r win10-x86 -c release /p:ShowLinkerSizeComparison=true
`
and get those errors.

`dotnet restore`
does not have issues and the project can be published/run just fine otherwise.

Note: you have to add latest ILLink.Tasks 0.1.5-preview-1841731 to AutoMasshTik.UI.csproj.

_Copied from original issue: dotnet/coreclr#19896_

**Comment by karelz** [comment]
_From @Petermarcu on September 11, 2018 20:19_

I moved this hear because its sounds like an interaction with the ILLink tasks.

**Comment by karelz** [comment]
@sbomer @jeffschwMSFT should it go to https://github.com/mono/linker repo?

**Comment by karelz** [comment]
_From @sbomer on January 18, 2019 20:56_

@karelz I think it should go in mono/linker.

**Comment by karelz** [comment]
@sbomer thanks! Feel free to move it yourself in future if that's the case ;)

**Comment by vanashimko** [comment]
@karelz faced similar behavior with NU1202, netcoreapp and netstandard, but in my case was able to solve by changing `TargetFrameworks` to `TargetFrameowrk` (no `s` at the end) in referenced netstandard2.0 project file. Looking through the repo you linked can't see any `TargetFrameworks` there though, so not sure about root cause.

Published my example here: https://github.com/vanashimko/dotnet-illink-issue. I've stripped out all the fat from my csproj files, so it can be easily seen, that removing `s` solves the problem. What I've noticed is that using TargetFrameworks with single framework produces slightly different project.assets.json, where crossTargeting flag is set to true (it is the only difference). Maybe the problem is somewhere there.
